src/containers/CreatePipelineRun/YAMLEditor.test.js

```diff
@@ -212,3 +239,158 @@ describe('YAMLEditor', () => {
     });
   });
 });
+
+describe('YAML. Edit and Run', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.spyOn(window.history, 'pushState');
+    // Workaround for codemirror vs jsdom https://github.com/jsdom/jsdom/issues/3002#issuecomment-1118039915
+    // for textRange(...).getClientRects is not a function
+    Range.prototype.getBoundingClientRect = () => ({
+      bottom: 0,
+      height: 0,
+      left: 0,
+      right: 0,
+      top: 0,
+      width: 0
+    });
+    Range.prototype.getClientRects = () => ({
+      item: () => null,
+      length: 0,
+      [Symbol.iterator]: jest.fn()
+    });
```

Comment:
We already have this in at least one other test, can it be extracted to an appropriately named function for reuse?

Reply:
after rework it is not needed